packages/node.fetcher/src/getNodeTarball.ts

```diff
@@ -8,8 +8,10 @@ export function getNodeTarball (
 ) {
   const platform = processPlatform === 'win32' ? 'win' : processPlatform
   const arch = normalizeArch(processPlatform, processArch)
+  const nodeMajorVersion = +nodeVersion.split('.')[0]
+  const nodeBinaryArch = (platform === 'darwin' && arch === 'arm64' && (nodeMajorVersion < 16)) ? 'x64' : arch
```

Comment:
I think you should make this changes in this file: https://github.com/pnpm/pnpm/blob/main/packages/node.fetcher/src/normalizeArch.ts